src/WebInterface.js

```diff
@@ -0,0 +1,105 @@
+const util = require('util');
+const fs   = require('fs');
+const jwt  = require('jsonwebtoken');
+
+// Set up an Express app for the Web API.
+const express    = require('express');
+const bodyParser = require('body-parser');
+const app        = express();
+const router     = express.Router();
+app.use(bodyParser.urlencoded({ extended: true }));
+
+const Config = require('./Config');
+
+class WebInterface {
+  
+  constructor(state) {
+    this.state = state;
+    this.port  = Config.get('webPort') || 9000;
+  }
+
+  init() {
+
+    const message = { message: "RanvierMUD API Endpoints: /npcs /players /items /rooms /help" };
+    // Routes for the API's GET response.
+    router.get('/',        (req, res) => res.json(message));
+    router.get('/npcs',    this.getResponseData(this.state.MobFactory, 'npcs'));
+    router.get('/players', this.getResponseData(this.state.PlayerManager, 'players'));     
+    router.get('/items',   this.getResponseData(this.state.ItemManager, 'items'));
+    router.get('/rooms',   this.getResponseData(this.state.RoomManager, 'rooms'));
+    router.get('/help',    this.getResponseData(this.state.HelpManager, 'helps'));
+
+    router.post('/authenticate', (req, res) => {
+      const accountName = req.body.name;
+      const account = this.state.AccountManager.findByName(accountName);
+      
+      if (!account) {
+        util.log(`[WEB] ${accountName} Auth Failed: Account not found.`);
+        return res.json({ success: false, message: `${accountName} was not found.`});
+      }
+
+      // Both passwords should be hashed... right?
```

Comment:
Do we really want to send passwords in plaintext? I would think it would be hashed on the client side and sent over as a hash, using the same algorithm.